### PR TITLE
Fix Cross compilation

### PR DESCRIPTION
**Pull request by liberodark**
https://github.com/mamedev/mame/pull/7670

### CODE REVIEW

File: minimp3.h [hunk 1]
```diff
@@ -889,7 +896,7 @@ static void L3_midside_stereo(float *left, int n)
         VSTORE(right + i, VSUB(vl, vr));
     }
 #endif /* HAVE_SIMD */
-    for (; i < n; i++)
+    i = (n >> 2) << 2;
```
Comment (lieff) [inline minimp3.h:899]: This change is not correct. Tests failing.

**Comment by lieff** [comment]
What is SBC? Some compiler? Can you give link for it?

**Comment by liberodark** [comment]
SBC is boards like Raspberry Pi 4

**Review by lieff — CHANGES_REQUESTED** [review]

**Comment by lieff** [comment]
Is toolchain for this board publicly available?
Can't guarantee code will not break again unless compiler included in test procedure.

File: minimp3.h (hunk 1 above)
Comment (liberodark) [reply minimp3.h:899]: Im go to see that.

**Comment by liberodark** [comment]
Yes send you information tonight about this

**Comment by lieff** [comment]
So patch basically defaults HAVE_SSE=0 and undefs it in HAVE_SSE=1 branch.
I suspect you run into build system issue rather code bug. Here current simd enabling branches:
```
#if !defined(MINIMP3_NO_SIMD)

#if (defined(_MSC_VER) && (defined(_M_IX86) || defined(_M_X64))) || ((defined(__i386__) || defined(__x86_64__)) && defined(__SSE2__))
#define HAVE_SSE 1
#define HAVE_SIMD 1
#elif defined(__ARM_NEON) || defined(__aarch64__) || defined(_M_ARM64)
#define HAVE_SSE 0
#define HAVE_SIMD 1
#else /* SIMD checks... */
#define HAVE_SSE 0
#define HAVE_SIMD 0
#endif /* SIMD checks... */

#else /* !defined(MINIMP3_NO_SIMD) */

#define HAVE_SIMD 0

#endif /* !defined(MINIMP3_NO_SIMD) */
```
Only branch you need set HAVE_SSE to zero is x86/x64 sse branch `#if (defined(_MSC_VER) && (defined(_M_IX86) || defined(_M_X64))) || ((defined(__i386__) || defined(__x86_64__)) && defined(__SSE2__))`
I assume code mistakenly go to this branch so you need reset HAVE_SSE=0 to make HAVE_SIMD=1 path work.
But root of the issue that HAVE_SIMD=1 path should not be taken.

Can you show compile error happen without code modification? And toolchain link so I can try it by myself.

**Comment by liberodark** [comment]
Issue is with this file : https://github.com/mamedev/mame/blob/master/3rdparty/minimp3/minimp3.h

**Comment by lieff** [comment]
Simd enabling code there same as upstream: sse branch condition is `#if (defined(_MSC_VER) && (defined(_M_IX86) || defined(_M_X64))) || ((defined(__i386__) || defined(__x86_64__)) && defined(__SSE2__))`
It can be mistakenly taken if build system wrongly set  _MSC_VER + _M_IX86 or __i386__ + __SSE2__ for example.
Need to see error message or toolchain to find out more.

**Comment by lieff** [comment]
Another possibility that HAVE_SSE is externally set. For example
```
$ gcc -DHAVE_SSE=0 minimp3_test.c -lm                                                                                                ↑
In file included from minimp3_ex.h:9,
                 from minimp3_test.c:40:
minimp3.h:99: warning: "HAVE_SSE" redefined
   99 | #define HAVE_SSE 1
      |

```
gives warning, so -Werror will fail.

**Pull request closed without merging**